airflow/providers/cncf/kubernetes/operators/kubernetes_pod.py

```diff
@@ -324,3 +310,97 @@ def _set_name(self, name):
             return None
         validate_key(name, max_length=63)
         return re.sub(r'[^a-z0-9.-]+', '-', name.lower())
+
+    def create_new_pod_for_operator(self, labels, launcher) -> Tuple[State, k8s.V1Pod, Optional[str]]:
+        '''
+        Creates a new pod and monitors for duration of task
+
+        @param labels:
+        @param launcher:
+        @return:
+        '''
+        if not (self.full_pod_spec or self.pod_template_file):
+            # Add Airflow Version to the label
+            # And a label to identify that pod is launched by KubernetesPodOperator
+            self.labels.update(
+                {
+                    'airflow_version': airflow_version.replace('+', '-'),
+                    'kubernetes_pod_operator': 'True',
+                }
+            )
+            self.labels.update(labels)
+        pod = pod_generator.PodGenerator(
+            image=self.image,
+            namespace=self.namespace,
+            cmds=self.cmds,
+            args=self.arguments,
+            labels=self.labels,
+            name=self.name,
+            envs=self.env_vars,
+            extract_xcom=self.do_xcom_push,
+            image_pull_policy=self.image_pull_policy,
+            node_selectors=self.node_selectors,
+            annotations=self.annotations,
+            affinity=self.affinity,
+            image_pull_secrets=self.image_pull_secrets,
+            service_account_name=self.service_account_name,
+            hostnetwork=self.hostnetwork,
+            tolerations=self.tolerations,
+            configmaps=self.configmaps,
+            security_context=self.security_context,
+            dnspolicy=self.dnspolicy,
+            schedulername=self.schedulername,
+            init_containers=self.init_containers,
+            restart_policy='Never',
+            priority_class_name=self.priority_class_name,
+            pod_template_file=self.pod_template_file,
+            pod=self.full_pod_spec,
+        ).gen_pod()
+
+        # noinspection PyTypeChecker
+        pod = append_to_pod(
+            pod,
+            self.pod_runtime_info_envs +  # type: ignore
+            self.ports +  # type: ignore
+            self.resources +  # type: ignore
+            self.secrets +  # type: ignore
+            self.volumes +  # type: ignore
+            self.volume_mounts  # type: ignore
+        )
+        try:
+            (final_state, result) = launcher.run_pod(
+                pod,
+                startup_timeout=self.startup_timeout_seconds,
+                get_logs=self.get_logs)
+        except AirflowException:
+            if self.log_events_on_failure:
+                for event in launcher.read_pod_events(pod).items:
+                    self.log.error("Pod Event: %s - %s", event.reason, event.message)
+            raise
+        finally:
+            if self.is_delete_operator_pod:
+                launcher.delete_pod(pod)
+
+        if final_state != State.SUCCESS:
```

Comment:
The event fetching should live in monitor_launched_pod, otherwise if we "attach"/monitor an existing pod we'll never check events for it.

Reply:
Ideally also can we change how we call launcher so that this fn (`create_new_pod_for_operator`) can call `monitor_launched_pod` -- so that both paths (creats vs attach) have the same behavoiur?

(Right now it appears the l.run_pod is blocking?)

Reply:
This probably means splitting `run_pod` inside launcher to be (pseudo code)

```python
pod = create_pod(...)
return monitor(pod)
```